### PR TITLE
Merchant stuff

### PR DESCRIPTION
**Pull request by Spummm**

### CODE REVIEW

File: code/datums/trading/misc.dm [hunk 1]
```diff
@@ -36,13 +36,17 @@
...
+								/mob/living/simple_animal/slime       = TRADER_THIS_TYPE,
+								/mob/living/simple_animal/hostile/giant_spider      = TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/beast/diyaab = TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/bear= TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/beast/shantak= TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/parrot      = TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/beast/samak= TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/goat = TRADER_THIS_TYPE,
-								/mob/living/simple_animal/hostile/carp = TRADER_THIS_TYPE)
+								/mob/living/simple_animal/hostile/carp = TRADER_All,
```
Comment (Kaksisilma) [inline code/datums/trading/misc.dm]: This needs to be capitalized
```suggestion
								/mob/living/simple_animal/hostile/carp = TRADER_ALL,
```

File: code/datums/trading/misc.dm [hunk 2]
```diff
@@ -56,13 +60,16 @@
...
+								/mob/living/simple_animal/slime       = TRADER_THIS_TYPE,
+								/mob/living/simple_animal/hostile/giant_spider      = TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/beast/diyaab = TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/bear= TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/beast/shantak= TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/parrot      = TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/beast/samak= TRADER_THIS_TYPE,
 								/mob/living/simple_animal/hostile/retaliate/goat = TRADER_THIS_TYPE,
-								/mob/living/simple_animal/hostile/carp= TRADER_THIS_TYPE,
+								/mob/living/simple_animal/hostile/carp= TRADER_All,
```
Comment (Kaksisilma) [inline code/datums/trading/misc.dm]:
```suggestion
								/mob/living/simple_animal/hostile/carp= TRADER_ALL,
```

**Review by Kaksisilma — CHANGES_REQUESTED** [review]

File: code/datums/trading/misc.dm (hunk 1 above)
Comment (Spummm) [reply code/datums/trading/misc.dm]: math is hard

**Review by Spummm — COMMENTED** [review]
e

**Review by Spummm — COMMENTED** [review]
yes

**Comment by Kaksisilma** [comment]
Looks good

**Pull request merged**